packages-content-model/roosterjs-content-model-plugins/lib/autoFormat/ContentModelAutoFormatPlugin.ts

```diff
@@ -0,0 +1,87 @@
+import keyboardListTrigger from './keyboardListTrigger';
+import type {
+    EditorPlugin,
+    IStandaloneEditor,
+    KeyDownEvent,
+    PluginEvent,
+} from 'roosterjs-content-model-types';
+import type { IContentModelEditor } from 'roosterjs-content-model-editor';
+
+/**
+ * Auto Format plugin handles auto formatting, such as transforming * characters into a bullet list.
+ * It can be customized with options to enable or disable auto list features.
+ * @param options An optional parameter that takes in an object of type AutoFormatOptions, which includes the following properties:
+ *  - autoBullet: A boolean that enables or disables automatic bullet list formatting. Defaults to true.
+ *  - autoNumbering: A boolean that enables or disables automatic numbering formatting. Defaults to true.
+ */
+export class ContentModelAutoFormatPlugin implements EditorPlugin {
+    private editor: IContentModelEditor | null = null;
+    private options = {
+        autoBullet: true,
+        autoNumbering: true,
+    };
+
+    constructor(options?: { autoBullet: boolean; autoNumbering: boolean }) {
```

Comment:
Suggest move the parameter out to be a type so that it is easier to add comment, and add more features later.

Also I think we need a default value for options

Reply:
For example:

```ts
const DefaultOptions: Required<YourTypeName> = {
  autoBullet: true;
  autoNumbering: true;
};

class ContentModelAutoFormatPlugin {
  constructor(private options: YouTypeName = DefaultOptions) { ... }
}
```

When you use this.options, no need to do null check.